metadata/containers.go

```diff
@@ -61,7 +61,64 @@ func (s *containerStore) Get(ctx context.Context, id string) (containers.Contain
 		return containers.Container{}, errors.Wrapf(err, "failed to read container %q", id)
 	}
 
-	return container, nil
+	if len(fieldmasks) == 0 {
+		return container, nil
+	}
+
+	ctrFieldMasked := new(containers.Container)
+	for _, path := range fieldmasks {
```

Comment:
This may be already implemented somewhere in the filters package. It is slightly different, but there should be common logic there. If not, we should make some utilities to do this consistently across the API.